app/stores/UsersStore.js

```diff
@@ -66,35 +68,46 @@ export default class UsersStore extends BaseStore<User> {
 
   @action
   promote = (user: User) => {
-    this.counts.admins += 1;
-    if (user.isViewer) this.counts.viewers -= 1;
-    return this.actionOnUser("promote", user);
+    try {
+      this.updateCounts(Rank.ADMIN, getUserRank(user));
+      this.actionOnUser("promote", user);
```

Comment:
Applies to all of these updates – need to `await` `this.actionOnUser` otherwise this `catch` will never be applicable.

Reply:
oh my gosh! how did I miss this!